runtime/caml/misc.h

```diff
@@ -42,8 +42,9 @@
   #define CAMLdeprecated_typedef(name, type) typedef type name
 #endif
 
-#if defined(__GNUC__) && __STDC_VERSION__ >= 199901L \
- || defined(_MSC_VER) && _MSC_VER >= 1925
+#if defined(__GNUC__)                                           \
+    && defined(__STDC_VERSION__) && __STDC_VERSION__ >= 199901L \
```

Comment:
Argh!  `__STDC_VERSION__` is **always** defined, the C standards say so.  No `defined` test, please.

Reply:
Is it always defined if the header is being read in C++, though?

Reply:
https://en.cppreference.com/w/cpp/preprocessor/replace says that `__STDC_VERSION__` may be defined by a C++ compiler, but doesn't say it must...

Note that this block of macros is here to define the `CAML_DEPRECATED` macro, which is no longer used in OCaml 5, as we've removed everything marked `CAML_DEPRECATED` in OCaml 4.  So, another way to go about it is to remove all these macros :-)

Reply:
Actually, we have a few `CAML_DEPRECATED` left, that's strange.  At some point maybe we'll get rid of that...

Reply:
> https://en.cppreference.com/w/cpp/preprocessor/replace says that `__STDC_VERSION__` may be defined by a C++ compiler, but doesn't say it must…

Neither GCC, clang, nor MSVC define it in C++ mode.

> Actually, we have a few `CAML_DEPRECATED` left, that's strange. At some point maybe we'll get rid of that…

We could keep it around to deprecate other things in the future.